src/main/kotlin/com/github/biomejs/intellijbiome/lsp/BiomeLspServerSupportProvider.kt

```diff
@@ -24,39 +24,36 @@ import kotlin.io.path.Path
         file: VirtualFile,
         serverStarter: LspServerSupportProvider.LspServerStarter,
     ) {
-        val currentConfigPath = project.service<BiomeServerService>().getCurrentConfigPath()
-        if (currentConfigPath != null) {
-            val executable = BiomePackage(project).binaryPath(currentConfigPath, file, false) ?: return
-            serverStarter.ensureServerStarted(BiomeLspServerDescriptor(project, executable, currentConfigPath))
-            return
-        }
+        // Finds the root directory of a Biome workspace. It's typically the parent directory of `biome.json`.
+        // If no `biome.json` file found, fallbacks to the project root directory.
+        val projectRootDir = project.guessProjectDir() ?: return
+        val root = file.findNearestBiomeConfig(projectRootDir)?.parent ?: projectRootDir
+
+        val configPath = BiomePackage(project).configPath()
+        val executable = BiomePackage(project).binaryPath(root.path, file, false) ?: return
 
-        val configPath = BiomePackage(project).configPath(file)
-        val executable = BiomePackage(project).binaryPath(configPath, file, false) ?: return
-        serverStarter.ensureServerStarted(BiomeLspServerDescriptor(project, executable, configPath))
+        serverStarter.ensureServerStarted(BiomeLspServerDescriptor(project, root, executable, configPath))
     }
 
     override fun createLspServerWidgetItem(lspServer: LspServer,
         currentFile: VirtualFile?) =
         LspServerWidgetItem(lspServer, currentFile, BiomeIcons.BiomeIcon, BiomeConfigurable::class.java)
 }
 
-@Suppress("UnstableApiUsage") private class BiomeLspServerDescriptor(project: Project,
+@Suppress("UnstableApiUsage") private class BiomeLspServerDescriptor(
+    project: Project,
+    private val root: VirtualFile,
     executable: String,
-    configPath: String?) : ProjectWideLspServerDescriptor(project, "Biome") {
+    configPath: String?,
+) : LspServerDescriptor(project, "Biome", root) {
     private val targetRun = run {
         val params = SmartList<ProcessCommandParameter>(ProcessCommandParameter.Value("lsp-proxy"))
         if (!configPath.isNullOrEmpty()) {
             params.add(ProcessCommandParameter.Value("--config-path"))
             params.add(ProcessCommandParameter.FilePath(Path(configPath)))
         }
 
-        BiomeTargetRunBuilder(project).getBuilder(executable).apply {
-            if (configPath.isNullOrEmpty()) {
-                setWorkingDirectory(configPath)
-            }
```

Comment:
Setting null or empty working directory is not intended? Anyway, we don't need to set the working directory I believe.